### ISSUE TITLE
Incorrect autofix of string concat with `prefer-spread` rule

### ISSUE THREAD

**Issue by bmish** [sindresorhus/eslint-plugin-unicorn#1147]
eslint-plugin-unicorn v29

Seeing a false positive from the `prefer-spread` rule mistaking a string variable for an array. Perhaps the rule could ignore variables that are clearly initialized as strings. 

Original:

```js
let className = 'responsive-column product-page__media-block';
className = className.concat(' media-block__responsive-column--sixth');
```

After autofix:

```js
let className = 'responsive-column product-page__media-block';
className = [...className, ' media-block__responsive-column--sixth'];
```

**Comment by fisker** [comment]
> Perhaps the rule could ignore variables that are clearly initialized as strings.


Already did, https://github.com/sindresorhus/eslint-plugin-unicorn/blob/c5a43459e881d36e1aa79a7e41a0ec0a05b7e7b7/rules/prefer-spread.js#L326-L328

Need investigation.

**Comment by sindresorhus** [comment]
Sidenote: In general, using `.concat` for a string is an anti-pattern though. Just use `+`.

**Comment by bmish** [comment]
Agreed on that, this is just a code sample from my large codebase which I saw autofixed incorrectly.

Looks like `getStaticValue` returns `null` when the string variable uses `let`, so we can't use that function to tell what type the variable is in that situation.

**Comment by fisker** [comment]
I'm going to close this, since there is nothing we can do.
